runtime_config_linux.go

```diff
@@ -133,59 +133,59 @@ type ThrottleDevice struct {
 // BlockIO for Linux cgroup 'blkio' resource management
 type BlockIO struct {
 	// Specifies per cgroup weight, range is from 10 to 1000
-	Weight uint16 `json:"blkioWeight"`
+	Weight *uint16 `json:"blkioWeight,omitempty"`
 	// Specifies tasks' weight in the given cgroup while competing with the cgroup's child cgroups, range is from 10 to 1000, CFQ scheduler only
-	LeafWeight uint16 `json:"blkioLeafWeight"`
+	LeafWeight *uint16 `json:"blkioLeafWeight,omitempty"`
 	// Weight per cgroup per device, can override BlkioWeight
-	WeightDevice []*WeightDevice `json:"blkioWeightDevice"`
+	WeightDevice []*WeightDevice `json:"blkioWeightDevice,omitempty"`
 	// IO read rate limit per cgroup per device, bytes per second
-	ThrottleReadBpsDevice []*ThrottleDevice `json:"blkioThrottleReadBpsDevice"`
+	ThrottleReadBpsDevice []*ThrottleDevice `json:"blkioThrottleReadBpsDevice,omitempty"`
 	// IO write rate limit per cgroup per device, bytes per second
-	ThrottleWriteBpsDevice []*ThrottleDevice `json:"blkioThrottleWriteBpsDevice"`
+	ThrottleWriteBpsDevice []*ThrottleDevice `json:"blkioThrottleWriteBpsDevice,omitempty"`
 	// IO read rate limit per cgroup per device, IO per second
-	ThrottleReadIOPSDevice []*ThrottleDevice `json:"blkioThrottleReadIOPSDevice"`
+	ThrottleReadIOPSDevice []*ThrottleDevice `json:"blkioThrottleReadIOPSDevice,omitempty"`
 	// IO write rate limit per cgroup per device, IO per second
-	ThrottleWriteIOPSDevice []*ThrottleDevice `json:"blkioThrottleWriteIOPSDevice"`
+	ThrottleWriteIOPSDevice []*ThrottleDevice `json:"blkioThrottleWriteIOPSDevice,omitempty"`
 }
 
 // Memory for Linux cgroup 'memory' resource management
 type Memory struct {
-	// Memory limit (in bytes)
-	Limit uint64 `json:"limit"`
-	// Memory reservation or soft_limit (in bytes)
-	Reservation uint64 `json:"reservation"`
-	// Total memory usage (memory + swap); set `-1' to disable swap
-	Swap uint64 `json:"swap"`
-	// Kernel memory limit (in bytes)
-	Kernel uint64 `json:"kernel"`
+	// Memory limit (in bytes).
+	Limit *int64 `json:"limit,omitempty"`
+	// Memory reservation or soft_limit (in bytes).
+	Reservation *int64 `json:"reservation,omitempty"`
+	// Total memory limit (memory + swap).
+	Swap *int64 `json:"swap,omitempty"`
+	// Kernel memory limit (in bytes).
+	Kernel *int64 `json:"kernel,omitempty"`
 	// Kernel memory limit for tcp (in bytes)
-	KernelTCP uint64 `json:"kernelTCP"`
-	// How aggressive the kernel will swap memory pages. Range from 0 to 100. Set -1 to use system default
-	Swappiness uint64 `json:"swappiness"`
+	KernelTCP *int64 `json:"kernelTCP"`
+	// How aggressive the kernel will swap memory pages. Range from 0 to 100.
+	Swappiness *int64 `json:"swappiness,omitempty"`
```

Comment:
On Wed, Dec 16, 2015 at 09:03:47PM -0800, Aleksa Sarai wrote:
> >  	// Kernel memory limit for tcp (in bytes)
> > -	KernelTCP uint64 `json:"kernelTCP"`
> > -	// How aggressive the kernel will swap memory pages. Range from 0 to 100. Set -1 to use system default
> > -	Swappiness uint64 `json:"swappiness"`
> > +	KernelTCP *int64 `json:"kernelTCP"`
> > +	// How aggressive the kernel will swap memory pages. Range from 0 to 100.
> > +	Swappiness *int64 `json:"swappiness,omitempty"`
>
> I'd also like to point out that using `-1` would've been the wrong
> type given the diff (the old code used `uint64` and told you to use
> `-1` for "default").

Those have just been out of sync since #199 landed.  An earlier
version of this PR (0d03c3d4) was turning it back to a signed integer
to fix this, but since we're not consistently using pointers we don't
need to restore signedness.